src/main/java/net/minecraftforge/unification/Unified.java

```diff
@@ -0,0 +1,103 @@
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.unification;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Represents a set of interchangeable things, with one chosen to be the 'unified' variant.
+ * It is defined by a {@link UnificationMaterial}, a {@link UnificationForm}, and a list of variants.
+ *
+ * To get an instance, you can add your own with {@link UnificationRegistry#add(UnificationMaterial, UnificationForm, T)}.
+ * If you have no ore you can just retrieve one with {@link UnificationRegistry#get(UnificationMaterial, UnificationForm)}.
+ *
+ * One variant is selected to be "the one" which is output from {@link #get()}.
+ * The variant that gets selected can be manipulated by the Forge Config, see "forge.configgui.unificationRegistryPriorityModIds"
+ */
+public final class Unified<T> implements Supplier<T>
+{
+    private final List<T> variants = new ArrayList<>();
+    private final UnificationMaterial material;
+    private final UnificationForm form;
+    private final Function<T, T> outputCopier;
+
+    /**
+     * Create an instance by using {@link UnificationRegistry#add(UnificationMaterial, UnificationForm, T)}
+     */
+    Unified(UnificationMaterial material, UnificationForm form, Function<T, T> outputCopier)
+    {
+        this.material = material;
+        this.form = form;
+        this.outputCopier = outputCopier;
+    }
+
+    /**
+     * @return the chosen 'unified' variant
+     */
+    @Override
+    public T get()
+    {
+        T variant = variants.get(0);
+        return outputCopier.apply(variant);
+    }
+
+    /**
+     * @return all the variants that have been registered
+     */
+    public List<T> getVariants()
+    {
+        return Collections.unmodifiableList(variants);
+    }
+
+    /**
+     * @return the {@link UnificationMaterial} property
+     */
+    public UnificationMaterial getMaterial()
+    {
+        return material;
+    }
+
+    /**
+     * @return the {@link UnificationForm} property
+     */
+    public UnificationForm getForm()
+    {
+        return form;
+    }
+
+    /**
+     * Use {@link UnificationRegistry#add(UnificationMaterial, UnificationForm, T)} to add a variant
+     */
+    void addVariant(T variant, Comparator<T> sortOrder)
+    {
+        this.variants.add(variant);
+        this.variants.sort(sortOrder);
```

Comment:
If you keep sorting this, I'd prefer using a `TreeSet` which sorts input elements efficiently.

Reply:
Good idea, will do.